### PR TITLE
yahoospp bidder& aol bidder: GPP Support in bid requests

### PR DESCRIPTION
**Pull request by radubarbos**
## Type of change

- [ ] Bugfix
- [ X] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

File: modules/yahoosspBidAdapter.js [hunk 1]
```diff
@@ -184,46 +184,46 @@ function validateAppendObject(validationType, allowedKeys, inputObject, appendTo
...
         if (allowedKeys.indexOf(objectKey) !== -1 && isArray(inputObject[objectKey])) {
           outputObject[objectKey] = inputObject[objectKey];
-        };
+        }
         break;
       case 'object':
         if (allowedKeys.indexOf(objectKey) !== -1 && isPlainObject(inputObject[objectKey])) {
           outputObject[objectKey] = inputObject[objectKey];
-        };
+        }
```
Comment (patmmccann) [inline modules/yahoosspBidAdapter.js:203]: please keep all these semicolons, we plan to turn on a no implied semicolon linting rule in the future

File: modules/yahoosspBidAdapter.js [hunk 2]
```diff
@@ -277,21 +277,29 @@ function generateOpenRtbObject(bidderRequest, bid) {
       }
     } else {
       outBoundBidRequest.site.id = bid.params.dcn;
-    };
+    }
+
+    if (bidderRequest.gppConsent) {
+      outBoundBidRequest.regs.ext.gpp = bidderRequest.gppConsent.gppString;
+      outBoundBidRequest.regs.ext.gpp_sid = bidderRequest.gppConsent.applicableSections;
+    } else if (bidderRequest.ortb2?.regs?.gpp) {
```
Comment (patmmccann) [inline modules/yahoosspBidAdapter.js:285]: why doesnt aol have this same logic?

**Review by patmmccann — CHANGES_REQUESTED** [review]
In addition to the in line comments; why haven't you passed GPP on your user syncs?

File: modules/aolBidAdapter.js [hunk 3]
```diff
@@ -373,6 +374,11 @@ export const spec = {
       params.us_privacy = consentData.uspConsent;
     }
 
+    if (consentData.gppConsent && consentData.gppConsent.gppString) {
```
Comment (patmmccann) [inline modules/aolBidAdapter.js:377]: why not mirror the logic below from yahoo?

File: modules/yahoosspBidAdapter.js (hunk 1 above)
Comment (radubarbos) [reply modules/yahoosspBidAdapter.js]: addressed

File: modules/yahoosspBidAdapter.js (hunk 2 above)
Comment (radubarbos) [reply modules/yahoosspBidAdapter.js:285]: the aol bidder adapter is the old one we are migrating off. Also there we have undocumented support for openRTB config, we don't want to add this support in the legacy adapter.

File: modules/aolBidAdapter.js (hunk 3 above)
Comment (radubarbos) [reply modules/aolBidAdapter.js:377]: this is a legacy adapter sending the requests to some legacy endpoints as GET requests. The POST openRTB is not officially supported. We don't apply the openRTB logic here.

**Comment by radubarbos** [comment]
> In addition to the in line comments; why haven't you passed GPP on your user syncs?

The user sync pixel is provided by our back-end with all the needed params, it is provided in the response.body.bidResponse.ext.pixels field.

File: modules/aolBidAdapter.js (hunk 3 above)
Comment (patmmccann) [reply modules/aolBidAdapter.js:377]: The only current place you can expect to find the gpp string is in the location yahoo falls back to based on publisher setConfigs.

**Comment by patmmccann** [comment]
> 

https://github.com/prebid/Prebid.js/blob/1b42c9a6692fa68716b216dfd2f4d15da9fd7c7a/modules/yahoosspBidAdapter.js#L634 you pass yourself consent in the syncs function but don't act on it. How does your backend become aware of consent? This seems quite unusual

File: modules/yahoosspBidAdapter.js (hunk 2 above)
Comment (dgirardi) [reply modules/yahoosspBidAdapter.js:285]: if `bidderRequest.gppConsent` is defined, then the `ortb2.regs` properties are as well; you can save some bytes and complexity by looking only at the latter. 

In fact, you could replace a lot of the code in here (esp. the giant `appendFirstPartyData` function) with just `mergeDeep(outBoundBidRequest, bidderRequest.ortb2)`.

**Comment by radubarbos** [comment]
We are aware of this very small chance that we could use a stale consent, it has always worked like this, we'll revisit together with the back-end team how we could improve this.
How long can the user syncs pixels could be hold in memory before they are fired? how big this delay could be?

File: modules/aolBidAdapter.js (hunk 3 above)
Comment (radubarbos) [reply modules/aolBidAdapter.js:383]: I added the fallback support similar to the yahoo adapter logic.

**Comment by patmmccann** [comment]
This delay could easily be up to 60 seconds, as publishers have a configuration parameter allowing their delay. The USP spec, and GPP has been less clear, says consent must be rechecked from the cmp each time it is needed.

File: modules/yahoosspBidAdapter.js (hunk 2 above)
Comment (patmmccann) [reply modules/yahoosspBidAdapter.js:285]: https://github.com/prebid/Prebid.js/blob/d5746c39c437b5e04505cdd75bdef3791eb28566/modules/consentManagementGpp.js#L380

You can mergeDeep regs, site, app, and user

Comment (radubarbos) [reply modules/yahoosspBidAdapter.js]: I updated it to always use the ortb2 data. For the regs we can not mergeDeep, our backend supports openRTB 2.5 and the gpp under "regs.ext.gpp".

**Comment by radubarbos** [comment]
Thanks for the information. We'll discuss this with the back-end teams.

**Comment by patmmccann** [comment]
@radubarbos just confirming, we're waiting for the results of your internal discussion?

**Comment by radubarbos** [comment]
@patmmccann We would like to work on that as a different ticket/task. We still have to coordinate with the back-end to discuss. If there are no more comments on these particular changes we would liked them merged. 
This opinion is also shared by @slimkrazy

**Review by patmmccann — APPROVED** [review]

**Comment by patmmccann** [comment]
@radubarbos we haven't heard from you in a while. We're planning to disclose our concerns in your documentation https://github.com/prebid/prebid.github.io/pull/4529/files

**Pull request merged**